### ISSUE TITLE
Position info for BLangIf node is incorrect

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#26085]
Consider the following if-else statement
```ballerina
if (true) {
    // some code
} else {
    // some more code
}
```
The whole thing above is modeled using the `BLangIf` node. But the position of this node is set to the same position as the block stmt of the body of the if condition. This causes false negative results in the `isWithinBlock()` check in the NodeResolver.

**Comment by pubudu91** [comment]
The issue seems to be caused by this particular call: https://github.com/ballerina-platform/ballerina-lang/blob/4ef8320836c2902696ebd05b0acd2e4086e6bfb2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java#L2542

Prior to this, the position is correct.

**Comment by pubudu91** [comment]
Closing since this was fixed with #26072
